src/Solvers/batched_tridiagonal_solver.jl

```diff
@@ -73,48 +70,132 @@ Reference implementation per Numerical Recipes, Press et. al 1992 (§ 2.4).
 """
 function solve!(ϕ, solver::BatchedTridiagonalSolver, rhs, args... )
 
-    a, b, c, t, parameters = solver.a, solver.b, solver.c, solver.t, solver.parameters
-    grid = solver.grid
-
-    launch!(architecture(solver), grid, :xy,
-            solve_batched_tridiagonal_system_kernel!, ϕ, a, b, c, rhs, t, grid, parameters, Tuple(args))
+    launch_config = if solver.tridiagonal_direction == :x
+                        :yz
+                    elseif solver.tridiagonal_direction == :y
+                        :xz
+                    elseif solver.tridiagonal_direction == :z
+                        :xy
+                    end
+
+    launch!(architecture(solver), solver.grid, launch_config,
+            solve_batched_tridiagonal_system_kernel!, ϕ,
+            solver.a,
+            solver.b,
+            solver.c,
+            rhs,
+            solver.t,
+            solver.grid,
+            solver.parameters,
+            Tuple(args),
+            Val(solver.tridiagonal_direction))
 
     return nothing
 end
 
+@inline get_coefficient(a::AbstractArray{T, 1}, i, j, k, grid, p, ::Val{:x},   args...) where {T} = @inbounds a[i]
+@inline get_coefficient(a::AbstractArray{T, 1}, i, j, k, grid, p, ::Val{:y},   args...) where {T} = @inbounds a[j]
+@inline get_coefficient(a::AbstractArray{T, 1}, i, j, k, grid, p, ::Val{:z},   args...) where {T} = @inbounds a[k]
+@inline get_coefficient(a::AbstractArray{T, 3}, i, j, k, grid, p, tridiag_dir, args...) where {T} = @inbounds a[i, j, k]
+
+@inline get_coefficient(a::Base.Callable, i, j, k, grid, p,         tridiagonal_direction, args...) = a(i, j, k, grid, p, args...)
+@inline get_coefficient(a::Base.Callable, i, j, k, grid, ::Nothing, tridiagonal_direction, args...) = a(i, j, k, grid, args...)
+
 @inline float_eltype(ϕ::AbstractArray{T}) where T <: AbstractFloat = T
 @inline float_eltype(ϕ::AbstractArray{<:Complex{T}}) where T <: AbstractFloat = T
 
-@kernel function solve_batched_tridiagonal_system_kernel!(ϕ, a, b, c, f, t, grid, p, args)
-    Nx, Ny, Nz = grid.Nx, grid.Ny, grid.Nz
+@kernel function solve_batched_tridiagonal_system_kernel!(ϕ, a, b, c, f, t, grid, p, args, tridiagonal_direction::Val{:x})
+    N = size(grid, 1)
+    m, n = @index(Global, NTuple)
```

Comment:
```suggestion
    Nx = size(grid, 1)
    i, j = @index(Global, NTuple)
```

this makes the code easier to read, right?